util/file.py

```diff
@@ -82,6 +82,28 @@ def get_resources():
         resources = json.load(inf)
     return resources
 
+def check_paths(read=(), write=(), read_and_write=()):
+    '''Check that we can read and write the specified files, throw an exception if not.  Useful for checking
+    error conditions early in the execution of a command.  Each arg can be a filename or iterable of filenames.
+    '''
+    read, write, read_and_write = map(util.misc.make_seq, (read, write, read_and_write))
```

Comment:
This passes in Py3? I thought `map` returned an iterable in Py3 (and `util.misc.make_seq` would then be lazily executed later on) and the multiple reads of the iterator down below would fail?

Reply:
map returns iterable over (make_seq(read), make_seq(write), make_seq(read_and_write))
then tuple unpacking runs through that iterable once,, assigning read=make_seq(read),, write=make_seq(write) etc
make_seq returns a tuple

Reply:
Ah okay, I didn't read that carefully. Looks good to merge!